graphql-dgs/src/main/kotlin/com/netflix/graphql/dgs/federation/DefaultDgsFederationResolver.kt

```diff
@@ -70,22 +71,25 @@ open class DefaultDgsFederationResolver() :
                 if (!fetcher.second.parameterTypes.any { it.isAssignableFrom(Map::class.java) }) {
                     throw InvalidDgsEntityFetcher("@DgsEntityFetcher ${fetcher.first::class.java.name}.${fetcher.second.name} is invalid. A DgsEntityFetcher must accept an argument of type Map<String, Object>")
                 }
+                try {
+                    val result =
+                        if (fetcher.second.parameterTypes.any { it.isAssignableFrom(DgsDataFetchingEnvironment::class.java) }) {
+                            fetcher.second.invoke(fetcher.first, values, DgsDataFetchingEnvironment(env))
+                        } else {
+                            fetcher.second.invoke(fetcher.first, values)
+                        }
+
+                    if (result == null) {
+                        throw MissingDgsEntityFetcherException(typename.toString())
+                    }
 
-                val result =
-                    if (fetcher.second.parameterTypes.any { it.isAssignableFrom(DgsDataFetchingEnvironment::class.java) }) {
-                        fetcher.second.invoke(fetcher.first, values, DgsDataFetchingEnvironment(env))
+                    if (result is CompletionStage<*>) {
+                        result.toCompletableFuture()
                     } else {
-                        fetcher.second.invoke(fetcher.first, values)
+                        CompletableFuture.completedFuture(result)
                     }
-
-                if (result == null) {
-                    throw MissingDgsEntityFetcherException(typename.toString())
-                }
-
-                if (result is CompletionStage<*>) {
-                    result.toCompletableFuture()
-                } else {
-                    CompletableFuture.completedFuture(result)
+                } catch (e: InvocationTargetException) {
+                    throw e.targetException
```

Comment:
Should it check if targetException is not null? I'm thinking about the case where the arguments of the method are of the wrong type. In that case the actual exception *is* `InvocationTargetException`. Maybe handle that specifically?

Reply:
Good call, will fix.